packages/adapter-knex/lib/adapter-knex.js

```diff
@@ -680,8 +680,13 @@ class QueryBuilder {
   }
 
   _getQueryConditionByPath(listAdapter, path, tableAlias) {
-    const dbPath = path.split('_', 1);
-    const fieldAdapter = listAdapter.fieldAdaptersByPath[dbPath];
+    let fieldAdapter = listAdapter.fieldAdaptersByPath[path];
+
+    if (!fieldAdapter) {
+      let dbPath = path.split('_').slice(0, -1).join('_');
```

Comment:
```suggestion
      const dbPath = path.split('_').slice(0, -1).join('_');
```

Reply:
Is it a lint rule that we use `let` unless it is redefined? If yes, can we enforce this as a lint rule?

Reply:
Not a rule at the moment, but definitely one I'd like, as that's what we aim for in practice 👍